### PR TITLE
Add schema property value check for all template files

### PR DESCRIPTION
**Pull request by zjpjack**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

File: test/solution-template-validation-tests/test/mainTemplateTests.js [hunk 1]
```diff
@@ -30,6 +30,19 @@ function getErrorMessage(obj, file, message) {
 /** Tests for template files in a solution template */
 describe('template files - ', () => {
 
+    /** Tests for $schema property in template files */
+    describe('$schema property tests - ', () => {
+        /** schema property MUST use https  */
+        it('schema property must use https not http', () => {
+            templateFileJSONObjects.forEach(templateJSONObject => {
+                var templateObject = templateJSONObject.value;
+                if (templateObject.$schema) {
```
Comment (vivsriaus) [inline test/solution-template-validation-tests/test/mainTemplateTests.js]: Ensure they all have schema property, just like createUiDef

File: test/solution-template-validation-tests/test/mainTemplateTests.js [hunk 2]
```diff
@@ -30,6 +30,19 @@ function getErrorMessage(obj, file, message) {
...
+                    templateObject.$schema.should.withMessage('in file:' + templateJSONObject.filename + ',$schema property must use https not http').contain('https');
```
Comment (vivsriaus) [inline test/solution-template-validation-tests/test/mainTemplateTests.js]: Ensure full string equals:
http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#

File: test/solution-template-validation-tests/test/mainTemplateTests.js (hunk 1 above)
Comment (zjpjack) [reply test/solution-template-validation-tests/test/mainTemplateTests.js]: done

File: test/solution-template-validation-tests/test/mainTemplateTests.js (hunk 2 above)
Comment (zjpjack) [reply test/solution-template-validation-tests/test/mainTemplateTests.js]: corrected

File: test/solution-template-validation-tests/test/mainTemplateTests.js (hunk 1 above)
Comment (vivsriaus) [reply test/solution-template-validation-tests/test/mainTemplateTests.js]: Not required, since we check for the entire schema value string above

Comment (bmoore-msft) [reply test/solution-template-validation-tests/test/mainTemplateTests.js]: how do we know that this file is a template file?  and not just a json file?

Comment (zjpjack) [reply test/solution-template-validation-tests/test/mainTemplateTests.js]: In util.js, we have function to get all template files, only the json files schema match "schema.management.azure.com/schemas/(.*)/deploymentTemplate.json", we consider the json files as template files.

Comment (bmoore-msft) [reply test/solution-template-validation-tests/test/mainTemplateTests.js]: That won't catch the case where the schema is missing...  We'll need to use something else, for example, the top level properties that are required (I think resources it the only one).

**Comment by MSidda** [comment]
i need to check ARM templates using travis ci ..please provide ..travis.yml file for checking parameters validation

**Pull request closed without merging**